src/Groups/MatGrp.jl

```diff
@@ -0,0 +1,888 @@
+order(::Type{T}, x::MatrixGroupElem)  where T<:Number = T(GAP.Globals.Order(x.X))
+
+#FIXME for the following functions, the output may not belong to the parent group of x
+#=
+frobenius(x::MatrixGroupElem, n::Int) = MatrixGroupElem(x.parent, matrix(x.parent.ring, x.parent.deg, x.parent.deg, [frobenius(y,n) for y in x.elm]))
+frobenius(x::MatrixGroupElem) = frobenius(x,1)
+
+transpose(x::MatrixGroupElem) = MatrixGroupElem(x.parent, transpose(x.elm))
+=#
+
+########################################################################
+#
+# Methods on groups
+#
+########################################################################
+
+"""
+    base_ring(G::MatrixGroup)
+
+Return the base ring of the matrix group `G`.
+"""
+base_ring(G::MatrixGroup) = G.ring
+
+"""
+    degree(G::MatrixGroup)
+
+Return the degree of the matrix group `G`, i.e. the number of rows of its matrices.
+"""
+degree(G::MatrixGroup) = G.deg
+
+Base.one(G::MatrixGroup) = MatrixGroupElem(G, identity_matrix(G.ring, G.deg))
+
+function Base.rand(G::MatrixGroup)
+   x_gap = GAP.Globals.Random(G.X)
+   x_oscar = G.mat_iso(x_gap)
+   return MatrixGroupElem(G,x_oscar,x_gap)
+end
+
+function gens(G::MatrixGroup)
+   isdefined(G,:gens) && return G.gens
+   
+   L = GAP.Globals.GeneratorsOfGroup(G.X)
+   G.gens=[MatrixGroupElem(G,G.mat_iso(L[i]),L[i]) for i in 1:length(L)]
+   return G.gens
+end
+
+gen(G::MatrixGroup, i::Int) = gens(G)[i]
+
+Base.getindex(G::MatrixGroup, i::Int) = gen(G, i)
+
+ngens(G::MatrixGroup) = length(gens(G))
+
+function order(G::MatrixGroup)
+   isdefined(G, :order) && return G.order
+
+   G.order = fmpz(BigInt(GAP.Globals.Size(G.X)))
+   return G.order
+end
+
+########################################################################
+#
+# Constructors
+#
+########################################################################
+
+"""
+    general_linear_group(n::Int, q::Int)
+    general_linear_group(n::Int, F::FqNmodFiniteField)
+    GL = general_linear_group
+
+Return the general linear group of dimension `n` either over the field `F` or the field `GF(q)`.
+"""
+function general_linear_group(n::Int, F::Ring)
+   G = MatrixGroup(n,F)
+   G.descr = :GL
+   return G
+end
+
+function general_linear_group(n::Int, q::Int)
+   (a,b) = ispower(q)
+   if !isprime(b) throw(ArgumentError("The field size must be a prime power")) end
+   return general_linear_group(n, GF(b,a)[1])
+end
+
+function general_linear_group(::Type{T}, n::Int, q::Int) where T <: GAPGroup
+   if T<:PermGroup   return T(GAP.Globals.GL(_gap_filter(T), n, q))
+   elseif T<:MatrixGroup   return general_linear_group(n,q)
+   else throw(ArgumentError("Type not supported"))
+   end
+end
+
+function general_linear_group(::Type{T}, n::Int, F::Ring) where T <: GAPGroup
+   if T<:PermGroup   return T(GAP.Globals.GL(_gap_filter(T), n, Int(order(F))))
+   elseif T<:MatrixGroup   return general_linear_group(n,F)
+   else throw(ArgumentError("Type not supported"))
+   end
+end
```

Comment:
Are these really useful? I don't recall ever having used `GL(IsPermGroup, n, q)` in GAP (if at all, I'd have used the image of `IsomorphismPermGroup(GL(n,q,))`.

Thing is, the methods as you wrote them above have serious issues, in that they accept crazy stuff like `general_linear_group(MatrixGroup{Integers}, 3, 3)`. While one could fix that (e.g. by using `T <: MatrixGroup && return T(general_linear_group(n,F))`) I am wondering whether it is worth the effort?

This is a good example for my rule of thumb, "if you have no usecase for code, don't write it, it'll be wrong"

Reply:
We have written these when `MatrixGroup` was still a "normal" GAPGroup type. The methods `isomorphic_perm_group` and similar make these constructors useless. Do I delete them?